quarkus-amazon-services fails to build

### Describe the bug

building quarkus-amazon-services fails in in io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest

### Expected behavior

The build should complete successfully

### Actual behavior

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.198 s <<< FAILURE! -- in io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest
[ERROR] io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest -- Time elapsed: 6.198 s <<< ERROR!
java.lang.RuntimeException: 
io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor#config threw an exception: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:375)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:292)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:698)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor#config threw an exception: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:139)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:371)
	... 3 more
Caused by: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

```

### How to Reproduce?

1. Check out quarkus-amazon-services from github
2. Ensure docker is running
3. build the project using `mvn clean install -Dquickly -Dskip.gradle.tests -Dskip.gradle.build`

This results in the error above and:
```
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Quarkus - Amazon Services - Parent 999-SNAPSHOT:
[INFO] 
[INFO] Quarkus - Amazon Services - Parent ................. SUCCESS [  0.487 s]
[INFO] Quarkus - Amazon Services - BOM Internal ........... SUCCESS [  0.008 s]
[INFO] Quarkus - Amazon Services - BOM .................... SUCCESS [  0.006 s]
[INFO] Quarkus - Amazon Services - Build Parent ........... SUCCESS [  0.006 s]
[INFO] Quarkus - Amazon Services - Code Generator ......... SUCCESS [  2.114 s]
[INFO] Quarkus - Amazon Services - Code Generator Maven Plugin SUCCESS [  1.006 s]
[INFO] Quarkus - Amazon Services - Common ................. SUCCESS [  0.006 s]
[INFO] Quarkus - Amazon Services - Common - Runtime Spi ... SUCCESS [  0.634 s]
[INFO] Quarkus - Amazon Services - Apache Client - Runtime - Internal SUCCESS [  0.208 s]
[INFO] Quarkus - Amazon Services - Netty Client - Runtime - Internal SUCCESS [  0.234 s]
[INFO] Quarkus - Amazon Services - CRT .................... SUCCESS [  0.006 s]
[INFO] Quarkus - Amazon Services - CRT - Runtime .......... SUCCESS [  0.048 s]
[INFO] Quarkus - Amazon Services - CRT Client - Runtime - Internal SUCCESS [  0.182 s]
[INFO] Quarkus - Amazon Services - OpenTelemetry - Runtime - Internal SUCCESS [  0.044 s]
[INFO] Quarkus - Amazon Services - Common - Runtime ....... SUCCESS [  0.440 s]
[INFO] Quarkus - Amazon Services - Common - Deployment DevServices Spi SUCCESS [  0.109 s]
[INFO] Quarkus - Amazon Services - Common - Deployment Spi  SUCCESS [  0.093 s]
[INFO] Quarkus - Amazon Services - Apache Client - Internal - Deployment SUCCESS [  0.082 s]
[INFO] Quarkus - Amazon Services - Netty Client - Internal - Deployment SUCCESS [  0.091 s]
[INFO] Quarkus - Amazon Services - CRT - Deployment ....... SUCCESS [  0.093 s]
[INFO] Quarkus - Amazon Services - CRT Client - Internal - Deployment SUCCESS [  0.081 s]
[INFO] Quarkus - Amazon Services - Apache Client - Internal - Deployment SUCCESS [  0.016 s]
[INFO] Quarkus - Amazon Services - Common - Deployment .... SUCCESS [  0.277 s]
[INFO] Quarkus - Amazon Services .......................... SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - ACM .................... SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - ACM - Runtime .......... SUCCESS [  0.429 s]
[INFO] Quarkus - Amazon DevServices ....................... SUCCESS [  0.007 s]
[INFO] Quarkus - Amazon DevServices - ACM ................. SUCCESS [  0.091 s]
[INFO] Quarkus - Amazon Services - API Gateway ............ SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - Api Gateway - Runtime .. SUCCESS [  0.280 s]
[INFO] Quarkus - Amazon DevServices - Apigateway .......... SUCCESS [  0.079 s]
[INFO] Quarkus - Amazon Services - Bedrock ................ SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - Bedrock - Runtime ...... SUCCESS [  0.261 s]
[INFO] Quarkus - Amazon DevServices - Bedrock ............. SUCCESS [  0.092 s]
[INFO] Quarkus - Amazon Services - Bedrock Runtime ........ SUCCESS [  0.005 s]
[INFO] Quarkus - Amazon Services - Bedrock Runtime - Runtime SUCCESS [  0.228 s]
[INFO] Quarkus - Amazon DevServices - Bedrock Runtime ..... SUCCESS [  0.074 s]
[INFO] Quarkus - Amazon Services - Cognito User Pools ..... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - Cognito User Pools - Runtime SUCCESS [  0.232 s]
[INFO] Quarkus - Amazon DevServices - Cognito User Pools .. SUCCESS [  0.095 s]
[INFO] Quarkus - Amazon Services - CloudWatch ............. SUCCESS [  0.005 s]
[INFO] Quarkus - Amazon Services - CloudWatch - Runtime ... SUCCESS [  0.240 s]
[INFO] Quarkus - Amazon DevServices - CloudWatch .......... SUCCESS [  0.074 s]
[INFO] Quarkus - Amazon Services - CloudWatch Logs ........ SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - CloudWatch Logs - Runtime SUCCESS [  0.240 s]
[INFO] Quarkus - Amazon DevServices - CloudWatch Logs ..... SUCCESS [  0.090 s]
[INFO] Quarkus - Amazon Services - DynamoDB ............... SUCCESS [  0.006 s]
[INFO] Quarkus - Amazon Services - DynamoDB - Runtime ..... SUCCESS [  0.249 s]
[INFO] Quarkus - Amazon DevServices - DynamoDB ............ SUCCESS [  0.074 s]
[INFO] Quarkus - Amazon Services - ECR .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - ECR - Runtime .......... SUCCESS [  0.228 s]
[INFO] Quarkus - Amazon DevServices - ECR ................. SUCCESS [  0.115 s]
[INFO] Quarkus - Amazon Services - EventBridge ............ SUCCESS [  0.006 s]
[INFO] Quarkus - Amazon Services - EventBridge - Runtime .. SUCCESS [  0.264 s]
[INFO] Quarkus - Amazon DevServices - EventBridge ......... SUCCESS [  0.072 s]
[INFO] Quarkus - Amazon Services - IAM .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - IAM - Runtime .......... SUCCESS [  0.255 s]
[INFO] Quarkus - Amazon DevServices - IAM ................. SUCCESS [  0.088 s]
[INFO] Quarkus - Amazon Services - Kinesis ................ SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - Kinesis - Runtime ...... SUCCESS [  0.239 s]
[INFO] Quarkus - Amazon DevServices - Kinesis ............. SUCCESS [  0.071 s]
[INFO] Quarkus - Amazon Services - KMS .................... SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - KMS - Runtime .......... SUCCESS [  0.233 s]
[INFO] Quarkus - Amazon DevServices - KMS ................. SUCCESS [  0.071 s]
[INFO] Quarkus - Amazon Services - Lambda ................. SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - Lambda - Runtime ....... SUCCESS [  0.247 s]
[INFO] Quarkus - Amazon DevServices - Lambda .............. SUCCESS [  0.071 s]
[INFO] Quarkus - Amazon Services - RDS .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - RDS - Runtime .......... SUCCESS [  0.304 s]
[INFO] Quarkus - Amazon DevServices - RDS ................. SUCCESS [  0.075 s]
[INFO] Quarkus - Amazon Services - Resource Groups Tagging API SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - Resource Groups Tagging API - Runtime SUCCESS [  0.217 s]
[INFO] Quarkus - Amazon DevServices - Apigateway .......... SUCCESS [  0.069 s]
[INFO] Quarkus - Amazon Services - S3 ..................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - S3 - Runtime ........... SUCCESS [  0.249 s]
[INFO] Quarkus - Amazon DevServices - S3 .................. SUCCESS [  0.088 s]
[INFO] Quarkus - Amazon Services - EventBridge Scheduler .. SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - EventBridge Scheduler - Runtime SUCCESS [  0.221 s]
[INFO] Quarkus - Amazon DevServices - EventBridge Scheduler SUCCESS [  0.085 s]
[INFO] Quarkus - Amazon Services - Secrets Manager ........ SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - Secrets Manager - Runtime SUCCESS [  0.214 s]
[INFO] Quarkus - Amazon DevServices - Secrets Manager ..... SUCCESS [  0.069 s]
[INFO] Quarkus - Amazon Services - SES .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - SES - Runtime .......... SUCCESS [  0.221 s]
[INFO] Quarkus - Amazon DevServices - SES ................. SUCCESS [  0.070 s]
[INFO] Quarkus - Amazon Services - SESv2 .................. SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - SESv2 - Runtime ........ SUCCESS [  0.234 s]
[INFO] Quarkus - Amazon DevServices - SESv2 ............... SUCCESS [  0.070 s]
[INFO] Quarkus - Amazon Services - SFN .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - SFN - Runtime .......... SUCCESS [  0.224 s]
[INFO] Quarkus - Amazon DevServices - SFN ................. SUCCESS [  0.069 s]
[INFO] Quarkus - Amazon Services - SNS .................... SUCCESS [  0.002 s]
[INFO] Quarkus - Amazon Services - SNS - Runtime .......... SUCCESS [  0.220 s]
[INFO] Quarkus - Amazon DevServices - SNS ................. SUCCESS [  0.072 s]
[INFO] Quarkus - Amazon Services - SQS .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - SQS - Runtime .......... SUCCESS [  0.239 s]
[INFO] Quarkus - Amazon DevServices - SQS ................. SUCCESS [  0.087 s]
[INFO] Quarkus - Amazon Services - SSM .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - SSM - Runtime .......... SUCCESS [  0.266 s]
[INFO] Quarkus - Amazon DevServices - SSM ................. SUCCESS [  0.071 s]
[INFO] Quarkus - Amazon Services - STS .................... SUCCESS [  0.003 s]
[INFO] Quarkus - Amazon Services - STS - Runtime .......... SUCCESS [  0.221 s]
[INFO] Quarkus - Amazon DevServices - STS ................. SUCCESS [  0.069 s]
[INFO] Quarkus - Amazon Services - Cognito User Pools - Deployment SUCCESS [  2.784 s]
[INFO] Quarkus - Amazon Services - DynamoDB Enhanced ...... SUCCESS [  0.004 s]
[INFO] Quarkus - Amazon Services - DynamoDB Enhanced - Runtime SUCCESS [  0.222 s]
[INFO] Quarkus - Amazon Services - DynamoDB - Deployment .. SUCCESS [05:15 min]
[INFO] Quarkus - Amazon Services - DynamoDB Enhanced - Deployment SUCCESS [  3.609 s]
[INFO] Quarkus - Amazon Services - EventBridge Scheduler - Deployment SUCCESS [  2.029 s]
[INFO] Quarkus - Amazon Services - S3 - Deployment ........ FAILURE [ 19.235 s]

```

### Output of `uname -a` or `ver`

Darwin Bruces-MacBook-Pro.local 25.1.0 Darwin Kernel Version 25.1.0: Mon Oct 20 19:34:05 PDT 2025; root:xnu-12377.41.6~2/RELEASE_ARM64_T6041 arm64

### Output of `java -version`

openjdk version "23.0.1" 2024-10-15 OpenJDK Runtime Environment (build 23.0.1+11-39) OpenJDK 64-Bit Server VM (build 23.0.1+11-39, mixed mode, sharing)

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.10 (5f519b97e944483d878815739f519b2eade0a91d)

### Additional information

_No response_

Just fyi, I checked this project out for the first time yesterday, so there are no historic artifacts on my system. Per your suggestion @geoand :
```
[INFO] Running io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest
2025-11-06 13:11:22,334 INFO  [io.qua.dev.dep.com.ComposeRunner] (build-2) Compose is running command: docker compose up -d
2025-11-06 13:11:22,411 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  busybox Pulling 
2025-11-06 13:11:23,409 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  busybox Pulled 
2025-11-06 13:11:23,450 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  Network quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx_default  Creating
2025-11-06 13:11:23,464 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  Network quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx_default  Created
2025-11-06 13:11:23,467 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Creating
2025-11-06 13:11:23,488 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Created
2025-11-06 13:11:23,497 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Starting
2025-11-06 13:11:23,557 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18111)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Started
2025-11-06 13:11:23,561 INFO  [io.qua.dev.dep.com.ComposeRunner] (build-2) Compose has finished running
2025-11-06 13:11:23,571 WARN  [io.qua.dev.dep.com.ComposeDevServicesProcessor] (build-2) Could not start successfully Compose dev services for project quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx, reason: Cannot read the array length because "array" is null
2025-11-06 13:11:23,574 INFO  [io.qua.dev.dep.com.ComposeRunner] (build-2) Compose is running command: docker compose down -v -t 1
2025-11-06 13:11:23,631 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18116)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Stopping
2025-11-06 13:11:24,719 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18116)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Stopped
2025-11-06 13:11:24,719 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18116)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Removing
2025-11-06 13:11:24,726 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18116)  Container quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx-busybox-1  Removed
2025-11-06 13:11:24,727 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18116)  Network quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx_default  Removing
2025-11-06 13:11:24,930 INFO  [io.qua.dev.dep.com.ComposeRunner] (process-error-consumer-1-"docker"-18116)  Network quarkus-devservices-quarkus-amazon-s3-deployment-zxdewx_default  Removed
2025-11-06 13:11:24,937 INFO  [io.qua.dev.dep.com.ComposeRunner] (build-2) Compose has finished running
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.007 s <<< FAILURE! -- in io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest
[ERROR] io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest -- Time elapsed: 3.007 s <<< ERROR!
java.lang.RuntimeException: 
io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor#config threw an exception: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:375)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:292)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:698)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor#config threw an exception: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:139)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:371)
	... 3 more
Caused by: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

[INFO] Running io.quarkiverse.amazon.s3.deployment.S3CustomCredentialsProviderMissedBeanNameNamedClientTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.383 s -- in io.quarkiverse.amazon.s3.deployment.S3CustomCredentialsProviderMissedBeanNameNamedClientTest
[INFO] Running io.quarkiverse.amazon.s3.deployment.S3CustomCredentialsProviderTest
2025-11-06 13:11:25,710 INFO  [io.quarkus] (main) quarkus-amazon-s3-deployment 999-SNAPSHOT on JVM (powered by Quarkus 3.29.0) started in 0.445s. 
2025-11-06 13:11:25,710 INFO  [io.quarkus] (main) Profile test activated. 
2025-11-06 13:11:25,710 INFO  [io.quarkus] (main) Installed features: [amazon-sdk-s3, cdi]
2025-11-06 13:11:25,845 INFO  [io.quarkus] (main) quarkus-amazon-s3-deployment stopped in 0.001s
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.513 s -- in io.quarkiverse.amazon.s3.deployment.S3CustomCredentialsProviderTest
[INFO] Running io.quarkiverse.amazon.s3.deployment.S3DevServicesTest
2025-11-06 13:11:26,117 INFO  [tc.loc.8.1] (build-16) LOCALSTACK_HOST environment variable set to localhost (to match host-routable address for container)
2025-11-06 13:11:26,117 INFO  [tc.loc.8.1] (build-16) Creating container for image: localstack/localstack:4.8.1
2025-11-06 13:11:26,147 INFO  [tc.loc.8.1] (build-16) Container localstack/localstack:4.8.1 is starting: 4836f5cc1605b9e8499f2c1846002ecb612cefda5b8b231702eafe318aa734b0
2025-11-06 13:11:27,863 INFO  [tc.loc.8.1] (build-16) Container localstack/localstack:4.8.1 started in PT1.745941S
2025-11-06 13:11:28,035 INFO  [io.qua.ama.com.dep.DevServicesLocalStackProcessor] (build-16) Amazon Dev Services for localstack (s3) started. Other Quarkus applications in dev mode will find the LocalStack automatically.
2025-11-06 13:11:28,138 INFO  [io.quarkus] (main) quarkus-amazon-s3-deployment 999-SNAPSHOT on JVM (powered by Quarkus 3.29.0) started in 0.083s. 
2025-11-06 13:11:28,138 INFO  [io.quarkus] (main) Profile test activated. 
2025-11-06 13:11:28,138 INFO  [io.quarkus] (main) Installed features: [amazon-sdk-s3, cdi, compose]
2025-11-06 13:11:28,303 INFO  [io.quarkus] (main) quarkus-amazon-s3-deployment stopped in 0.001s
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.623 s -- in io.quarkiverse.amazon.s3.deployment.S3DevServicesTest
[INFO] Running io.quarkiverse.amazon.s3.deployment.S3CustomCredentialsProviderMissedBeanTest
2025-11-06 13:11:28,467 INFO  [io.qua.dev.com.ContainerShutdownCloseable] (main) Dev Services for localstack (s3) shut down.
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.372 s -- in io.quarkiverse.amazon.s3.deployment.S3CustomCredentialsProviderMissedBeanTest
[DEBUG] Closing the fork 1 after saying GoodBye.
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   DevServicesWithComposeDevServicesTest » Runtime io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor#config threw an exception: java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5466)
	at io.quarkus.devservices.deployment.compose.ComposeServiceWaitStrategyTarget.<init>(ComposeServiceWaitStrategyTarget.java:35)
	at io.quarkus.devservices.deployment.compose.ComposeProject.createServiceInstance(ComposeProject.java:317)
	at io.quarkus.devservices.deployment.compose.ComposeProject.discoverServiceInstances(ComposeProject.java:286)
	at io.quarkus.devservices.deployment.compose.ComposeProject.start(ComposeProject.java:178)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.startCompose(ComposeDevServicesProcessor.java:262)
	at io.quarkus.devservices.deployment.compose.ComposeDevServicesProcessor.config(ComposeDevServicesProcessor.java:122)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1575)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

[INFO] 
[ERROR] Tests run: 10, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.689 s
[INFO] Finished at: 2025-11-06T13:11:30Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.5.3:test (default-test) on project quarkus-amazon-s3-deployment: 
[ERROR] 
[ERROR] See /Users/brucej/development/quarkus-amazon-services/s3/deployment/target/surefire-reports for the individual test results.
[ERROR] See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.5.3:test (default-test) on project quarkus-amazon-s3-deployment: 

See /Users/brucej/development/quarkus-amazon-services/s3/deployment/target/surefire-reports for the individual test results.
See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoFailureException: 

See /Users/brucej/development/quarkus-amazon-services/s3/deployment/target/surefire-reports for the individual test results.
See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    at org.apache.maven.plugin.surefire.SurefireHelper.throwException (SurefireHelper.java:247)
    at org.apache.maven.plugin.surefire.SurefireHelper.reportExecution (SurefireHelper.java:159)
    at org.apache.maven.plugin.surefire.SurefireMojo.handleSummary (SurefireMojo.java:547)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:1147)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:969)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[DEBUG] Shutting down adapter factory; available factories [file-lock, rwlock-local, semaphore-local, noop]; available name mappers [discriminating, file-gav, file-hgav, file-static, gav, static]
[DEBUG] Shutting down 'file-lock' factory
[DEBUG] Shutting down 'rwlock-local' factory
[DEBUG] Shutting down 'semaphore-local
[DEBUG] Shutting down 'noop' factory
```

This PR is successful (for this issue) but fails to build other services which may be expected. For this issue using:
`mvn test -f s3/deployment/ -Dtest=DevServicesWithComposeDevServicesTest -Dquarkus.log.category.\"io.quarkus.devservices\".level=DEBUG`
I now get:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest
2025-11-06 16:16:16,046 INFO  [org.testcontainers.DockerClientFactory] (build-29) Testcontainers version: 1.21.3
2025-11-06 16:16:16,128 INFO  [org.testcontainers.dockerclient.DockerClientProviderStrategy] (build-29) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2025-11-06 16:16:16,236 INFO  [org.testcontainers.dockerclient.DockerClientProviderStrategy] (build-29) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
2025-11-06 16:16:16,237 INFO  [org.testcontainers.DockerClientFactory] (build-29) Docker host IP address is localhost
2025-11-06 16:16:16,241 INFO  [org.testcontainers.DockerClientFactory] (build-29) Connected to docker: 
  Server Version: 28.5.1
  API Version: 1.51
  Operating System: Docker Desktop
  Total Memory: 7836 MB
  Labels: 
    com.docker.desktop.address=unix:///Users/brucej/Library/Containers/com.docker.docker/Data/docker-cli.sock
2025-11-06 16:16:16,245 INFO  [org.testcontainers.images.PullPolicy] (build-29) Image pull policy will be performed by: DefaultPullPolicy()
2025-11-06 16:16:16,246 INFO  [org.testcontainers.utility.ImageNameSubstitutor] (build-29) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2025-11-06 16:16:16,248 INFO  [org.testcontainers.DockerClientFactory] (build-29) Checking the system...
2025-11-06 16:16:16,248 INFO  [org.testcontainers.DockerClientFactory] (build-29) ✔︎ Docker server version should be at least 1.6.0
2025-11-06 16:16:17,265 DEBUG [io.quarkus.devservices.deployment.compose.ComposeProject] (build-29) Added wait strategy org.testcontainers.containers.wait.strategy.HostPortWaitStrategy@277a6e2b for service busybox
2025-11-06 16:16:18,437 INFO  [tc.testcontainers/ryuk:0.12.0] (build-29) Creating container for image: testcontainers/ryuk:0.12.0
2025-11-06 16:16:18,502 INFO  [org.testcontainers.utility.RegistryAuthLocator] (build-29) Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
2025-11-06 16:16:18,549 INFO  [tc.testcontainers/ryuk:0.12.0] (build-29) Container testcontainers/ryuk:0.12.0 is starting: 9d8df2a0429f0d8a9c9e4953a86a69be30917ab8110b813f4aaeb260aab69b55
2025-11-06 16:16:18,675 INFO  [tc.testcontainers/ryuk:0.12.0] (build-29) Container testcontainers/ryuk:0.12.0 started in PT0.237709S
2025-11-06 16:16:18,678 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (build-29) Compose is running command: docker compose up -d
2025-11-06 16:16:18,735 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  busybox Pulling 
2025-11-06 16:16:19,872 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  busybox Pulled 
2025-11-06 16:16:19,919 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  Network quarkus-devservices-quarkus-amazon-s3-deployment-ixkrfd_default  Creating
2025-11-06 16:16:19,933 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  Network quarkus-devservices-quarkus-amazon-s3-deployment-ixkrfd_default  Created
2025-11-06 16:16:19,935 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  Container quarkus-devservices-quarkus-amazon-s3-deployment-ixkrfd-busybox-1  Creating
2025-11-06 16:16:19,958 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  Container quarkus-devservices-quarkus-amazon-s3-deployment-ixkrfd-busybox-1  Created
2025-11-06 16:16:19,967 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  Container quarkus-devservices-quarkus-amazon-s3-deployment-ixkrfd-busybox-1  Starting
2025-11-06 16:16:20,036 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (process-error-consumer-1-"docker"-25166)  Container quarkus-devservices-quarkus-amazon-s3-deployment-ixkrfd-busybox-1  Started
2025-11-06 16:16:20,040 INFO  [io.quarkus.devservices.deployment.compose.ComposeRunner] (build-29) Compose has finished running
2025-11-06 16:16:20,055 INFO  [io.quarkus.devservices.deployment.compose.ComposeProject] (build-2) Waiting for service busybox to be ready
2025-11-06 16:16:20,056 INFO  [io.quarkus.devservices.deployment.compose.ComposeProject] (build-2) Service busybox is ready
2025-11-06 16:16:20,064 INFO  [tc.localstack/localstack:4.8.1] (build-2) LOCALSTACK_HOST environment variable set to localhost (to match host-routable address for container)
2025-11-06 16:16:20,064 INFO  [tc.localstack/localstack:4.8.1] (build-2) Creating container for image: localstack/localstack:4.8.1
2025-11-06 16:16:20,253 INFO  [tc.localstack/localstack:4.8.1] (build-2) Container localstack/localstack:4.8.1 is starting: bac8497aff37afeb0918d42cbbd28cd936e9ba14ca6f182d3a46ff12073b7872
2025-11-06 16:16:21,262 INFO  [tc.localstack/localstack:4.8.1] (build-2) Container localstack/localstack:4.8.1 started in PT1.198227S
2025-11-06 16:16:21,722 INFO  [io.quarkiverse.amazon.common.deployment.DevServicesLocalStackProcessor] (build-2) Amazon Dev Services for localstack (s3) started. Other Quarkus applications in dev mode will find the LocalStack automatically.
2025-11-06 16:16:21,840 INFO  [io.quarkus] (main) quarkus-amazon-s3-deployment 999-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 0.069s. 
2025-11-06 16:16:21,841 INFO  [io.quarkus] (main) Profile test activated. 
2025-11-06 16:16:21,841 INFO  [io.quarkus] (main) Installed features: [amazon-sdk-s3, cdi, compose]
2025-11-06 16:16:21,921 INFO  [org.testcontainers.DockerClientFactory] (main) Testcontainers version: 1.21.3
2025-11-06 16:16:21,973 INFO  [org.testcontainers.dockerclient.DockerClientProviderStrategy] (main) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2025-11-06 16:16:22,039 INFO  [org.testcontainers.dockerclient.DockerClientProviderStrategy] (main) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
2025-11-06 16:16:22,040 INFO  [org.testcontainers.DockerClientFactory] (main) Docker host IP address is localhost
2025-11-06 16:16:22,049 INFO  [org.testcontainers.DockerClientFactory] (main) Connected to docker: 
  Server Version: 28.5.1
  API Version: 1.51
  Operating System: Docker Desktop
  Total Memory: 7836 MB
  Labels: 
    com.docker.desktop.address=unix:///Users/brucej/Library/Containers/com.docker.docker/Data/docker-cli.sock
2025-11-06 16:16:22,052 INFO  [org.testcontainers.images.PullPolicy] (main) Image pull policy will be performed by: DefaultPullPolicy()
2025-11-06 16:16:22,053 INFO  [org.testcontainers.utility.ImageNameSubstitutor] (main) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2025-11-06 16:16:22,115 INFO  [tc.testcontainers/ryuk:0.12.0] (main) Creating container for image: testcontainers/ryuk:0.12.0
2025-11-06 16:16:22,170 INFO  [org.testcontainers.utility.RegistryAuthLocator] (main) Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
2025-11-06 16:16:22,219 INFO  [tc.testcontainers/ryuk:0.12.0] (main) Container testcontainers/ryuk:0.12.0 is starting: b0dfe9e17baba592b08a0aed252a16fb40e400734659805b711c9b6080f63919
2025-11-06 16:16:22,325 INFO  [tc.testcontainers/ryuk:0.12.0] (main) Container testcontainers/ryuk:0.12.0 started in PT0.209399S
2025-11-06 16:16:22,326 INFO  [org.testcontainers.utility.RyukResourceReaper] (main) Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2025-11-06 16:16:22,326 INFO  [org.testcontainers.DockerClientFactory] (main) Checking the system...
2025-11-06 16:16:22,326 INFO  [org.testcontainers.DockerClientFactory] (main) ✔︎ Docker server version should be at least 1.6.0
2025-11-06 16:16:22,347 INFO  [io.quarkus] (main) quarkus-amazon-s3-deployment stopped in 0.002s
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 8.844 s -- in io.quarkiverse.amazon.s3.deployment.DevServicesWithComposeDevServicesTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.152 s
[INFO] Finished at: 2025-11-06T16:16:23Z
[INFO] ------------------------------------------------------------------------
```
...but as I say, it fails to build DynamoDB Enhanced - Deployment
```
[ERROR] /Users/brucej/development/quarkus-amazon-services/dynamodb-enhanced/deployment/src/main/java/io/quarkiverse/amazon/dynamodb/enhanced/deployment/DynamodbEnhancedDbTableProcessor.java:[162,66] incompatible types: io.quarkus.arc.processor.BeanConfiguratorBase.CreateGeneration cannot be converted to io.quarkus.gizmo.MethodCreator

```

--- a/extensions/devservices/common/src/main/java/io/quarkus/devservices/common/ContainerLocator.java
+++ b/extensions/devservices/common/src/main/java/io/quarkus/devservices/common/ContainerLocator.java
@@ -64,6 +64,10 @@ public class ContainerLocator {
     }
 
     private Optional<ContainerPort> getMappedPort(Container container, int port) {
+        if (container.getPorts() == null) {
+            return Optional.empty();
+        }
+
         return Arrays.stream(container.getPorts())
                 .filter(containerPort -> hasMatchingPort(containerPort, port))
                 .findAny();
@@ -99,6 +103,10 @@ public class ContainerLocator {
         if (shared && launchMode == LaunchMode.DEVELOPMENT) {
             return lookup(serviceName).findAny()
                     .map(container -> {
+                        if (container.getPorts() == null) {
+                            return container.getId();
+                        }
+
                         Arrays.stream(container.getPorts())
                                 .filter(cp -> Objects.nonNull(cp.getPublicPort()) && Objects.nonNull(cp.getPrivatePort()))
                                 .forEach(cp -> {

--- a/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/DevServicesProcessor.java
+++ b/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/DevServicesProcessor.java
@@ -349,6 +349,10 @@ public class DevServicesProcessor {
     }
 
     private ContainerInfo.ContainerPort[] getExposedPorts(Container container) {
+        if (container.getPorts() == null) {
+            return new ContainerInfo.ContainerPort[0];
+        }
+
         return Arrays.stream(container.getPorts())
                 .map(c -> new ContainerInfo.ContainerPort(c.getIp(), c.getPrivatePort(), c.getPublicPort(), c.getType()))
                 .toArray(ContainerInfo.ContainerPort[]::new);

--- a/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/compose/ComposeServiceWaitStrategyTarget.java
+++ b/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/compose/ComposeServiceWaitStrategyTarget.java
@@ -32,11 +32,11 @@ public class ComposeServiceWaitStrategyTarget implements WaitStrategyTarget, Sup
     public ComposeServiceWaitStrategyTarget(DockerClient dockerClient, Container container) {
         this.dockerClient = dockerClient;
         this.container = container;
-        this.exposedPorts = Arrays.stream(container.getPorts())
+        this.exposedPorts = container.getPorts() != null ? Arrays.stream(container.getPorts())
                 .filter(port -> port.getPublicPort() != null)
                 .map(ContainerPort::getPrivatePort)
                 .filter(Objects::nonNull)
-                .toList();
+                .toList() : List.of();
     }
 
     @Override

